src/index.ts

```diff
@@ -1,88 +1,106 @@
+      console.log('');
+      continue;
+    }
     packageScope.push(scope);
   }
-  console.log('');
-  const j1Client = await new JupiterOneClient({
-    account: 'j1dev',
-    username: 'j1dev',
-    password: 'dev',
-    poolId: process.env.J1_USER_POOL_ID,
-    clientId: process.env.J1_CLIENT_ID,
-    accessToken: process.env.J1_API_TOKEN,
-    dev: process.env.J1_DEV || false
-  }).init();
+  const pathToRepos = question('Input path to directory with repos (relative to root directory): ');
+  const account = question('Input account. If using a .env file, continue without input: ');
+  const username = question('Input username. If using a .env file, continue without input: ');
+  const password = question('Input password. If using a .env file, continue without input: ');
+  const access = question('Input access token. If using a .env file, continue without input: ');
+  const clientInput = {account: account, username: username, password: password, accessToken: access};
+
 
-  const repoMap = await getRepoIds();
+  console.log('');
+  const j1Client = await getClient(clientInput);
+  const repoMap = await getRepoIds(pathToRepos, clientInput);
+  console.log('');
 
-  for (const repoName of Array.from(repoMap.values())) {
+  for (const repoName of Array.from(repoMap.keys())) {
     console.log('Repo: ' + repoName);
 
     const mainRepo = await j1Client.queryV1(
-      `FIND CodeRepo WITH name='${repoName}'`,
+      `FIND CodeRepo WITH name='${repoName}'`
     );
-    const depsList = getDependencies(repoName, packageScope);
-    if (depsList === undefined) {
-      continue;
-    }
-    if (depsList.length > 0) {
-      for (const dep of depsList) {
-        const fullName = dep.substring(1, dep.indexOf(':'));
-        let depRepo = await j1Client.queryV1(
-          `FIND CodeRepo WITH full_name='${fullName}'`,
-        );
+    const depsList = getDependencies(repoName, packageScope, repoMap);
+    const deployDepsList = getDependenciesYaml(repoName, repoMap);
+    
+    switchName = '';
+    getSwitchExpression(depsList, deployDepsList);
 
-        if (depRepo.length > 0) {
-          const relationshipKey = mainRepo[0].entity._key + '|uses|' + depRepo[0].entity._key;
-          const relationship = await j1Client.createRelationship(
-            relationshipKey,
-            'bitbucket_repo_uses_bitbucket_repo',
-            'USES',
-            mainRepo[0].entity._id,
-            depRepo[0].entity._id,
-          );
-          console.log('Successfully created relationship (' + repoName + ' USES ' + dep + ').');
-          success++;
-        } else {
-          console.log(
-            'Failed to create relationship with ' + dep + 
-            ' (was not found on the graph). Skipped.'
-          );
-          failure++;
-          if (missingDeps.indexOf(dep) === -1) {
-            missingDeps.push(dep + ' (' + repoName + ').');
-          }
-        }
+    switch (switchName) {
```

Comment:
I think you can simplify a lot of this, just have two if / else blocks.

```
if (depsList !== undefined) {
    const relationship = await createRepoRelationship(depsList, mainRepo, j1Client, repoName, missingDeps);
    missingDeps = relationship.missingDeps;
    success += relationship.success;
    failure += relationship.failure;
} else {
    console.log('No package.json, not creating repo dependency relationships')
}

if (deployDepsList !== undefined) {
    const relationship = await createDeployRelationship(deployDepsList, mainRepo, j1Client, repoName, missingDeps);
    missingDeps = relationship.missingDeps;
    success += relationship.success;
    failure += relationship.failure;
} else {
    console.log('No dependencies.yaml, not creating deploy relationships');
}
```

Reply:
Bro whatcha got against switch statements